packages/realm-server/tests/helpers/index.ts

```diff
@@ -269,18 +263,27 @@ export async function createRealm({
   // if you are creating a realm  to test it directly without a server, you can
   // also specify `withWorker: true` to also include a worker with your realm
   withWorker?: true;
+  sourceFileDirectoryToCopy?: string;
 }): Promise<{ realm: Realm; adapter: RealmAdapter }> {
   await insertPermissions(dbAdapter, new URL(realmURL), permissions);
 
-  for (let [filename, contents] of Object.entries(fileSystem)) {
-    if (typeof contents === 'string') {
-      writeFileSync(join(dir, filename), contents);
-    } else {
-      writeJSONSync(join(dir, filename), contents);
+  let adapter = new NodeAdapter(dir, enableFileWatcher);
+
```

Comment:
we did this so we use `.write` as the main interface so file attributes are set inside of tests -- particular realm-server-tests